diff -ruN xf86-video-amdgpu-git/src/amdgpu_kms.c xf86-video-amdgpu-git-mod/src/amdgpu_kms.c
--- xf86-video-amdgpu-git/src/amdgpu_kms.c	2020-05-26 12:49:51.524700000 +0200
+++ xf86-video-amdgpu-git-mod/src/amdgpu_kms.c	2020-05-26 12:54:22.101383288 +0200
@@ -1681,8 +1681,8 @@
 		info->cursor_w = CURSOR_WIDTH;
 		info->cursor_h = CURSOR_HEIGHT;
 	} else {
-		info->cursor_w = CURSOR_WIDTH_CIK;
-		info->cursor_h = CURSOR_HEIGHT_CIK;
+		info->cursor_w = CURSOR_WIDTH;
+		info->cursor_h = CURSOR_HEIGHT;
 	}
 
 	amdgpu_query_heap_size(pAMDGPUEnt->pDev, AMDGPU_GEM_DOMAIN_GTT,
