diff -ruN mesa/include/pci_ids/radeonsi_pci_ids.h mesa-patched/include/pci_ids/radeonsi_pci_ids.h
--- mesa/include/pci_ids/radeonsi_pci_ids.h	2021-03-28 16:03:36.133569046 +0200
+++ mesa-patched/include/pci_ids/radeonsi_pci_ids.h	2021-03-28 16:08:04.963584560 +0200
@@ -88,6 +88,11 @@
 CHIPSET(0x665D, BONAIRE)
 CHIPSET(0x665F, BONAIRE)
 
+CHIPSET(0x9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS)
+
 CHIPSET(0x9830, KABINI)
 CHIPSET(0x9831, KABINI)
 CHIPSET(0x9832, KABINI)
diff -ruN mesa/src/amd/addrlib/src/amdgpu_asic_addr.h mesa-patched/src/amd/addrlib/src/amdgpu_asic_addr.h
--- mesa/src/amd/addrlib/src/amdgpu_asic_addr.h	2021-03-28 16:03:36.143569047 +0200
+++ mesa-patched/src/amd/addrlib/src/amdgpu_asic_addr.h	2021-03-28 16:31:38.140332659 +0200
@@ -71,7 +71,9 @@
 #define AMDGPU_HAWAII_RANGE     0x28, 0x3C
 
 #define AMDGPU_SPECTRE_RANGE    0x01, 0x41
-#define AMDGPU_SPOOKY_RANGE     0x41, 0x81
+#define AMDGPU_SPOOKY_RANGE     0x41, 0x61
+#define AMDGPU_STARSHA_RANGE    0x61, 0x71
+#define AMDGPU_STARSHP_RANGE    0x71, 0x81
 #define AMDGPU_KALINDI_RANGE    0x81, 0xA1
 #define AMDGPU_GODAVARI_RANGE   0xA1, 0xFF
 
@@ -123,6 +125,8 @@
 
 #define ASICREV_IS_SPECTRE(r)          ASICREV_IS(r, SPECTRE)
 #define ASICREV_IS_SPOOKY(r)           ASICREV_IS(r, SPOOKY)
+#define ASICREV_IS_STARSHA(r)           ASICREV_IS(r, STARSHA)
+#define ASICREV_IS_STARSHP(r)           ASICREV_IS(r, STARSHP)
 #define ASICREV_IS_KALINDI(r)          ASICREV_IS(r, KALINDI)
 #define ASICREV_IS_KALINDI_GODAVARI(r) ASICREV_IS(r, GODAVARI)
 
diff -ruN mesa/src/amd/common/ac_gpu_info.c mesa-patched/src/amd/common/ac_gpu_info.c
--- mesa/src/amd/common/ac_gpu_info.c	2021-03-28 16:03:36.156902381 +0200
+++ mesa-patched/src/amd/common/ac_gpu_info.c	2021-03-28 16:40:13.203695707 +0200
@@ -510,6 +510,8 @@
    case FAMILY_KV:
       identify_chip2(SPECTRE, KAVERI);
       identify_chip2(SPOOKY, KAVERI);
+      identify_chip2(STARSHA, LIVERPOOL);
+	  identify_chip2(STARSHP, GLADIUS);
       identify_chip2(KALINDI, KABINI);
       identify_chip2(GODAVARI, KABINI);
       break;
@@ -699,6 +701,8 @@
    case CHIP_VERDE:
    case CHIP_HAINAN:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_ICELAND:
    case CHIP_CARRIZO:
@@ -1226,6 +1230,8 @@
    case CHIP_PITCAIRN:
    case CHIP_VERDE:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_HAWAII:
    case CHIP_TONGA:
    case CHIP_FIJI:
@@ -1276,6 +1282,14 @@
       raster_config = 0x16000012;
       raster_config_1 = 0x00000000;
       break;
+   case CHIP_LIVERPOOL:
+      raster_config = 0x2a00161a;
+	  raster_config_1 = 0x00000000;
+	  break;
+   case CHIP_GLADIUS:
+	  raster_config = 0x2a00161a;
+	  raster_config_1 = 0x0000002e;
+	  break;
    /* 2 SEs / 8 RBs */
    case CHIP_TAHITI:
    case CHIP_PITCAIRN:
diff -ruN mesa/src/amd/common/amd_family.h mesa-patched/src/amd/common/amd_family.h
--- mesa/src/amd/common/amd_family.h	2021-03-28 16:03:36.160235714 +0200
+++ mesa-patched/src/amd/common/amd_family.h	2021-03-28 16:33:00.210337383 +0200
@@ -81,6 +81,8 @@
    CHIP_OLAND,
    CHIP_HAINAN,
    CHIP_BONAIRE, /* GFX7 (Sea Islands) */
+   CHIP_LIVERPOOL,
+   CHIP_GLADIUS,
    CHIP_KAVERI,
    CHIP_KABINI,
    CHIP_HAWAII,
diff -ruN mesa/src/amd/llvm/ac_llvm_util.c mesa-patched/src/amd/llvm/ac_llvm_util.c
--- mesa/src/amd/llvm/ac_llvm_util.c	2021-03-28 16:03:36.173569048 +0200
+++ mesa-patched/src/amd/llvm/ac_llvm_util.c	2021-03-28 16:43:30.537040445 +0200
@@ -133,6 +133,10 @@
       return "hainan";
    case CHIP_BONAIRE:
       return "bonaire";
+   case CHIP_LIVERPOOL:
+      return "bonaire";
+   case CHIP_GLADIUS:
+      return "bonaire";
    case CHIP_KABINI:
       return "kabini";
    case CHIP_KAVERI:
diff -ruN mesa/src/amd/vulkan/winsys/null/radv_null_winsys.c mesa-patched/src/amd/vulkan/winsys/null/radv_null_winsys.c
--- mesa/src/amd/vulkan/winsys/null/radv_null_winsys.c	2021-03-28 16:03:36.203569050 +0200
+++ mesa-patched/src/amd/vulkan/winsys/null/radv_null_winsys.c	2021-03-28 16:46:14.910383246 +0200
@@ -43,6 +43,10 @@
 	[CHIP_OLAND] = { 0x6060, 2, true },
 	[CHIP_HAINAN] = { 0x6660, 2, true },
 	[CHIP_BONAIRE] = { 0x6640, 4, true },
+	[CHIP_LIVERPOOL] = { 0x9920, 2, false },
+	[CHIP_LIVERPOOL] = { 0x9922, 2, false },
+	[CHIP_LIVERPOOL] = { 0x9923, 2, false },
+	[CHIP_GLADIUS] = { 0x9924, 4, false },
 	[CHIP_KAVERI] = { 0x1304, 2, false },
 	[CHIP_KABINI] = { 0x9830, 2, false },
 	[CHIP_HAWAII] = { 0x67A0, 16, true },
diff -ruN mesa/src/gallium/targets/d3dadapter9/description.c mesa-patched/src/gallium/targets/d3dadapter9/description.c
--- mesa/src/gallium/targets/d3dadapter9/description.c	2021-03-28 16:03:36.730235748 +0200
+++ mesa-patched/src/gallium/targets/d3dadapter9/description.c	2021-03-28 16:27:21.490317882 +0200
@@ -53,6 +53,8 @@
     {"KAVERI",                      "AMD Radeon(TM) R7 Graphics"},
     {"KABINI",                      "AMD Radeon HD 8400 / R3 Series"},
     {"BONAIRE",                     "AMD Radeon HD 8770"},
+    {"LIVERPOOL",                   "AMD Liverpool (Playstation 4)"},
+    {"GLADIUS",                     "AMD Gladius (Playstation 4)"},
     {"OLAND",                       "AMD Radeon HD 8670"},
     {"HAINAN",                      "AMD Radeon HD 8600M Series"},
     {"TAHITI",                      "AMD Radeon HD 7900 Series"},
diff -ruN mesa/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c mesa-patched/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
--- mesa/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c	2021-03-28 16:03:36.750235749 +0200
+++ mesa-patched/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c	2021-03-28 16:13:39.590270500 +0200
@@ -275,6 +275,8 @@
       ws->info.chip_class = GFX6;
       break;
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_KABINI:
    case CHIP_HAWAII:
@@ -496,6 +498,8 @@
    case CHIP_VERDE:
    case CHIP_OLAND:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.max_tcc_blocks = 4;
       break;
@@ -525,9 +529,11 @@
       case CHIP_TAHITI:
       case CHIP_PITCAIRN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
          ws->info.max_se = 2;
          break;
       case CHIP_HAWAII:
+      case CHIP_GLADIUS:
          ws->info.max_se = 4;
          break;
       }
